packages/cardano-services/src/Http/HttpService.ts

```diff
@@ -33,17 +33,17 @@ export abstract class HttpService extends RunnableModule {
   }
 
   async initializeImpl(): Promise<void> {
-    if (!(await this.healthCheck()).ok) {
-      throw new ProviderError(ProviderFailure.Unhealthy);
-    }
+    if (this.provider instanceof RunnableModule) await this.provider.initialize();
+
+    if (!(await this.healthCheck()).ok) throw new ProviderError(ProviderFailure.Unhealthy);
   }
 
   async startImpl(): Promise<void> {
-    await this.provider.start?.();
+    if (this.provider instanceof RunnableModule) await this.provider.start();
```

Comment:
I think the 2 approaches can be combined. `Runnable` as an interface is still a good idea here:
```suggestion
    function isRunnable(obj: unknown): obj is Runnable { return 'start' in obj && 'shutdown' in obj }
    if (isRunnable(this.provider)) await this.provider.start();
```